src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -303,26 +303,12 @@ internal bool CommitUnsynchronized()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal void Advance(int bytesWritten)
         {
-            if (_writingHead == null)
-            {
-                ThrowHelper.ThrowInvalidOperationException_NotWritingNoAlloc();
-            }
-
-            if (bytesWritten >= 0)
-            {
-                Debug.Assert(_writingHead.Next == null);
-
-                if (bytesWritten > _writingMemory.Length)
-                {
-                    ThrowHelper.ThrowInvalidOperationException_AdvancingPastBufferSize();
-                }
-
-                AdvanceCore(bytesWritten);
-            }
-            else
+            if (bytesWritten < 0 || bytesWritten > _writingMemory.Length)
```

Comment:
What happens now if _writingHead is null?

Reply:
And what about:
```C#
if ((uint)bytesWritten > (uint)_writingMemory.Length)
```
instead?

Reply:
If Advance(0) and writing head is null, it'll noop, if Advance non-zero and writing head is null, it'll throw an argument out of range. _writingMemory represents the amount of available space in the current segment.

Reply:
Done